### PR TITLE
Fix sphinx refs

### PR DESCRIPTION
**Pull request by fjetter**
# Description:

This enables intersphinx for our most used dependencies and fixes a lot of wrong links which were causing wrongfully rendered docs. The workaround with the target replacing is unfortunate but I couldn't find a nicer way to address the issue. For some reason the type annotations are not properly rendered. For example

`ParquetSerializer` would become `kartothek.serialization._parquet.ParquetSerializer` but the only available sphinx target is the public import without `_parquet`. Same is for the import redirects of dask, pandas, etc. but I believe this is fine

Partially, I also converted our doc style type annotations to proper annotations in some places. Covering the entire code base is out of scope for this PR

### CODE REVIEW

File: kartothek/serialization/_parquet.py [hunk 1]
```diff
@@ -99,18 +101,6 @@ def __init__(
...
-            isinstance(other, ParquetSerializer)
-            and (self.compression == other.compression)
-            and (self.chunk_size == other.chunk_size)
-        )
-
-    def __repr__(self):
-        return "ParquetSerializer(compression={compression!r}, chunk_size={chunk_size!r})".format(
-            compression=self.compression, chunk_size=self.chunk_size
-        )
-
```
Comment (fjetter) [inline kartothek/serialization/_parquet.py:113]: - [ ] oops

File: kartothek/io/dask/common_cube.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def build_cube_from_bag_internal(
 
     Parameters
     ----------
-    data:
+    data: dask.bag.Bag
```
Comment (stephan-hesselmann-by) [inline kartothek/io/dask/common_cube.py:104]: Why did you add the type here? Shouldn't it just be `data` (without colon) to be consistent with other changes (and numpy docstring style)?

Comment (fjetter) [reply kartothek/io/dask/common_cube.py:104]: yes, indeed. Sometimes my "regex search and replace" hammer was a bit too forceful, I assume. In particular with the cube API I had issues since the API is using a lot of type annotations in docs w/out having proper type annotations in general. Sometimes I fixed that, sometimes I didn't. TBH, I consider this PR one of the "it's not 100% but it still an improvement" things :)

If it is just about this line in particular, I can address this but I am affraid this or similar mistakes were made and I would prefer not to go over the code base again. This can be _eventually_ cleaned up since having the type documented twice should generally not be harmful

Comment (stephan-hesselmann-by) [reply kartothek/io/dask/common_cube.py:104]: Alright no problem, I just wanted to know if this was an intentional change.

**Review by stephan-hesselmann-by — APPROVED** [review]
